server/src/internalClusterTest/java/org/elasticsearch/index/mapper/IgnoredMetadataFieldIT.java

```diff
@@ -0,0 +1,164 @@
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.admin.indices.create.CreateIndexResponse;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.index.query.IdsQueryBuilder;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.aggregations.bucket.terms.StringTerms;
+import org.elasticsearch.search.aggregations.metrics.InternalAvg;
+import org.elasticsearch.test.ESSingleNodeTestCase;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.junit.Before;
+
+import static org.elasticsearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.avg;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.terms;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.hasSize;
+
+@SuppressWarnings("resource")
+public class IgnoredMetadataFieldIT extends ESSingleNodeTestCase {
+
+    public static final String NUMERIC_FIELD_NAME = "numeric_field";
+    public static final String DATE_FIELD_NAME = "date_field";
+    public static final String TEST_INDEX = "test-index";
+    public static final String CORRECT_FIELD_TYPE_DOC_ID = "1";
+    public static final String WRONG_FIELD_TYPE_DOC_ID = "2";
+
+    @Before
+    public void createTestIndex() throws Exception {
+        CreateIndexResponse createIndexResponse = null;
+        try {
+            XContentBuilder mapping = XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("_doc")
+                .startObject("properties")
+                .startObject(NUMERIC_FIELD_NAME)
+                .field("type", "long")
+                .field("ignore_malformed", true)
+                .endObject()
+                .startObject(DATE_FIELD_NAME)
+                .field("type", "date")
+                .field("ignore_malformed", true)
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject();
+            createIndexResponse = indicesAdmin().prepareCreate(TEST_INDEX).setMapping(mapping).get();
+            assertAcked(createIndexResponse);
+            indexTestDoc(NUMERIC_FIELD_NAME, CORRECT_FIELD_TYPE_DOC_ID, "42");
+            indexTestDoc(NUMERIC_FIELD_NAME, WRONG_FIELD_TYPE_DOC_ID, "forty-two");
+        } finally {
+            if (createIndexResponse != null) {
+                createIndexResponse.decRef();
+            }
+        }
+    }
+
+    public void testIgnoredMetadataFieldFetch() {
```

Comment:
This is a bit of a duplicate of MetadatFetchingIT#testWithIgnored ? Is it needed?

Reply:
I missed this comment...I can remove the test in another PR.